src/stacks-hierarchy/Root_TicketAssistantStack/TicketAssistant_DurationScreen/TicketAssistant_DurationScreen.tsx

```diff
@@ -0,0 +1,358 @@
+import {Platform, ScrollView, View} from 'react-native';
+import {StyleSheet} from '@atb/theme';
+import {themeColor} from '@atb/stacks-hierarchy/Root_TicketAssistantStack/TicketAssistant_WelcomeScreen';
+import {DashboardBackground} from '@atb/assets/svg/color/images';
+import {ThemeText} from '@atb/components/text';
+import {TicketAssistantTexts, useTranslation} from '@atb/translations';
+import {SliderComponent} from '@atb/components/slider';
+import {Button} from '@atb/components/button';
+import React, {useEffect, useState} from 'react';
+import {
+  sliderColorMax,
+  sliderColorMin,
+} from '@atb/stacks-hierarchy/Root_TicketAssistantStack/TicketAssistant_FrequencyScreen';
+import {TicketAssistantScreenProps} from '@atb/stacks-hierarchy/Root_TicketAssistantStack/navigation-types';
+import {useTicketAssistantState} from '@atb/stacks-hierarchy/Root_TicketAssistantStack/TicketAssistantContext';
+import RNDateTimePicker from '@react-native-community/datetimepicker';
+import {useLocaleContext} from '@atb/LocaleProvider';
+import {ThemeIcon} from '@atb/components/theme-icon';
+import SvgDate from '@atb/assets/svg/mono-icons/time/Date';
+import {format, parseISO} from 'date-fns';
+import {dateToDateString} from '@atb/components/sections/items/date-input/utils';
+import {useAccessibilityContext} from '@atb/AccessibilityContext';
+import {SectionSeparator} from '@atb/components/sections';
+import {
+  dateDiffInDays,
+  getDateFromSlider,
+  getDaysWeeksMonths,
+  getSliderIndex,
+} from './utils';
+type DurationProps =
+  TicketAssistantScreenProps<'TicketAssistant_DurationScreen'>;
+
+const currentDate = new Date();
+const durations = [1, 7, 14, 21, 30, 60, 90, 120, 150, 180];
+
+export const TicketAssistant_DurationScreen = ({navigation}: DurationProps) => {
+  const {data, updateData} = useTicketAssistantState();
+  const [date, setDate] = useState(dateToDateString(currentDate));
+  const [showDatePicker, setShowDatePicker] = useState(false);
+  const [usedSlider, setUsedSlider] = useState(false);
+  const styles = useThemeStyles();
+  const {t} = useTranslation();
+  const a11yContext = useAccessibilityContext();
+  const [sliderValue, setSliderValue] = useState(data.duration);
+
+  useEffect(() => {
+    const unsubscribe = navigation.addListener('blur', () => {
+      if (usedSlider) {
+        updateData({...data, duration: sliderValue});
+      }
+    });
```

Comment:
It updates data on `blur`? Is it the main flow, instead of doing it on slider change or pressing next, or is it necessary for some corner case?

Reply:
It is so that the value updates when you swipe, and not only when you press the next button